2nd PR: Front-End Code Optimization, UI Enhancements & Navigation Fixes

FrankFurter123 opened the pull request:
##  Overview  
This PR focuses on optimizing the website's structure, improving navigation, refining UI elements, and adding interactive features to enhance the overall user experience.  

---

##  Key Changes & Improvements  

### Code Refactoring & Optimization  
- Introduced a **reusable Header component** to streamline the website structure.  
- Removed unnecessary **Navbar CSS** for better integration.  
- Refactored `SignUp` and `DateOfBirth` components by removing redundant wrapper divs.  

### Navigation & Routing Fixes  
- Fixed **SignUp** and **Login** links to ensure they navigate to the correct pages.  
- Ensured that **Navbar and Footer do not appear** on **Login and SignUp** pages.  
- Connected **footer links** to their correct pages for seamless navigation.  
- Implemented **scroll-to-top on route change** in `App.jsx` to improve user experience.  
- Restored **Footer and Navbar** in `App.jsx` after accidental removal.  
- Added **"Join Now"** button in **Learn More** page linking to **Sign Up**.  

### UI/UX Enhancements  
- Applied **independent styles** for `About.css` and `Learn_More.css` to prevent global overrides.  
- Renamed `index.css` to `Home.css` for better clarity.  
- Improved **Home.jsx** sections by adding **titles** and **descriptive text**.  
- Introduced **interactive hover effects** for the **gallery images** to make them more engaging.  

---

## Testing & Validation  
✔️ Verified **proper routing** and **page redirections** for Login and SignUp.  
✔️ Ensured **Navbar and Footer do not render** on **Login and SignUp** pages.  
✔️ Tested **scroll-to-top functionality** on route changes.  
✔️ Confirmed **interactive gallery effects** work smoothly.  
✔️ Ensured **UI elements render correctly** across different screen sizes.  

---

## Notes  
Please take a closer look at the changes when reviewing our full code as I changed too many files for the optimization, and if there's anything else needed, let me know.  

I made this pull request to ensure we don’t run into issues later when we start implementing more complex websites, APIs, JavaScript, etc. I will make sure to do another optimization in the future.  

Some things I forgot to include were **comments for all of the code**, but I'll let you guys handle that as you go with the project.  

- Franklin La Rosa

AbdulAlharbi reviewed (commented):
### **Peer Review for PR #45 – Front-End Code Optimization, UI Enhancements & Navigation Fixes**  

#### **General Feedback**  
This PR brings a lot of improvements to the project's structure, navigation, and UI/UX. The refactoring efforts and modularization of components enhance maintainability, while the added interactive elements improve the overall user experience. Below are some observations, areas for improvement, and a few questions for discussion.  

---  

### **Strengths of the PR**  
**Refactored Code for Better Maintainability**  
- The introduction of a **reusable Header component** is a great step towards modularization.  
- Removing unnecessary **Navbar CSS** improves code organization and prevents redundant overrides.  
- Refactoring `SignUp` and `DateOfBirth` components by eliminating redundant wrapper divs makes the codebase cleaner.  

**Navigation & Routing Fixes**  
- Fixed **SignUp/Login** links and ensured they navigate to the correct pages.  
- Ensured **Navbar and Footer do not appear** on Login and SignUp pages, which improves the user experience.  
- Connected **footer links** properly to their corresponding pages.  
- Implemented **scroll-to-top on route change** in `App.jsx` to enhance navigation.  

**UI/UX Enhancements**  
- Applied **independent styles** for `About.css` and `Learn_More.css` to avoid global style conflicts.  
- Renamed `index.css` to `Home.css` for better clarity.  
- Improved **Home.jsx** by adding **titles & descriptive text** for better readability.  
- Introduced **interactive hover effects** for the **gallery section**, making it more engaging.  

---  

### **Areas for Improvement**  
**Reusability of Components**  
- Instead of manually adding/removing the Navbar and Footer in `App.jsx`, would using a **higher-order component (HOC) or a layout wrapper** be a cleaner approach to handling conditional rendering?  

**Code Comments & Documentation**  
- Since this PR introduces major structural changes, adding **comments in key areas** (such as the logic behind Navbar/Footer conditional rendering) would help other team members understand the changes better.  
- You mentioned you **forgot to include some comments**—would you be adding them in a future commit to ensure better clarity for other developers?  

---  

### **Questions for Discussion**  
1. **How can we further optimize the reusability of the Header and Footer components across different pages?** Would a Layout component or another method work better?  
2. **What was the reasoning behind renaming `index.css` to `Home.css`?** Was it mainly for clarity, or were there specific issues with conflicting styles?  

---  

### **Final Thoughts**  
This PR introduces **strong improvements** to the website's structure, navigation, and UI/UX. The refactoring efforts and interactive features are great additions. Addressing **component reusability** and adding **clearer comments/documentation** would make it even better.  

Once these minor refinements are considered, this PR will be even stronger! Let me know your thoughts on the points above.

FrankFurter123 commented:
Hey Abdul,

 I really appreciate you taking the time to review this! Thanks for the detailed feedback and for calling out both the improvements and areas to refine.
 
 To answer your 2 questions:

### Reusability of Components 
I implemented a Layout component to handle conditional rendering for the Navbar and Footer, keeping App.jsx cleaner. This way, any new pages that need the Header and Footer will have them automatically. The best part is that if a page doesn’t require them, we can easily add a conditional in App.jsx to exclude them. It took me a bit to figure out due to React’s structure, but it’s all working smoothly now! Here is a screenshot of the changes I made to the App.jsx and how to use the Footer and Header:

<img width="1039" alt="Screenshot 2025-03-10 at 12 56 59" src="https://github.com/user-attachments/assets/54f384b9-7d1e-4acc-bea3-73dda067567a" />

---

### Renaming index.css to Home.css 
This was mainly for clarity and organization. As you may know, I have been coding on CodeSanbox because I am more familiar with this platform instead of React and the default is index.html and index.css so when implementing my code in React I forgot to change it and to maintain consistency with out CSS styles and Jsx files I went ahead and changed the name.

---

For the Areas of improvements : Yes, I plan on adding comments to our code as well as any new code we implement. I’d also like you guys to start making this a habit moving forward—it’ll save us time in the long run!

Thanks again for the feedback Abdul!

- Franklin

AbdulAlharbi reviewed (dismissed):
Hey Frank,  

Thanks for the clarification! I appreciate the effort you put into making the **Header and Footer more reusable** with the Layout component—that's a solid improvement and definitely helps keep `App.jsx` cleaner. It’s great to see how you structured it to make future adjustments easier!  

Also, renaming `index.css` to `Home.css` for clarity makes perfect sense, and I totally get the reasoning behind it. Your attention to organization will definitely help maintain consistency across the project.  

I think you did a **pretty good job** so far! Everything looks well-structured, and I like that you're pushing for better commenting habits—it’ll definitely help in the long run.  

Approving this for now—**great job!** Keep up the good work!

Jose-Alarcon1 reviewed (approved):
**PR Review: Optimization**

You’ve made some solid improvements that cleans up the code. I went through the changes, and overall, everything looks great. Here’s my feedback:
________________________________________
**What’s Working Well**

1) Code Cleanup & Optimization
•	You introduced a reusable Header component. This makes the structure cleaner.
•	Removing unnecessary Navbar CSS and redundant wrapper divs was a smart move.
•	The SignUp and DateOfBirth components are more streamlined after your improvements.
2)  Navigation Fixes
•	The SignUp and Login links now work properly
•	Good call on making sure the Navbar and Footer don’t show up on Login and SignUp pages.
•	The scroll-to-top function is a great touch, makes navigation smoother.
3) UI/UX Improvements
•	Moving styles into separate files (About.css, Learn_More.css) helps keep things clean and avoid global conflicts.
•	Renaming index.css to Home.css makes it clearer what it’s for.
•	The hover effects on the gallery images add a nice interactive feel.
4) Testing & Validation
•	The key functionalities, routing, hiding/showing the Navbar and Footer, and scroll-to-top, have all been tested and work fine.
________________________________________
**Some Things to Improve**

1)	Missing Code Comments
•	A few comments here and there would help make future edits easier. Maybe drop them in a later PR.
2)	Big PR, Harder Review
•	There are a lot of changes packed into this PR. If possible, breaking future optimizations into smaller PRs might make it easier to track and review changes.
3)	More Testing?
•	Everything seems to work fine, but if we’re adding more complex features later, it’d be good to have some automated tests in place to prevent things from breaking.
4)	 Performance Check
•	It might be worth checking how assets impact load times.
________________________________________

Author: Jose Alarcon

FrankFurter123 commented:
Hey Abdul and Jose, I made a  last commit to set up the component structure, so you can just add your code to the respective files without needing to modify App.jsx, the footer, or the header. This should make integration smoother for everyone! It's ready to merge!

AbdulAlharbi reviewed (approved):
Approved.
